sdk/core/az_core/src/az_http_policy.c

```diff
@@ -70,11 +60,16 @@ AZ_NODISCARD az_result az_http_pipeline_policy_credential(
     _az_http_request* p_request,
     az_http_response* p_response)
 {
-  if (p_data != AZ_CREDENTIAL_ANONYMOUS)
+  _az_credential* const credential = (_az_credential*)p_data;
+  _az_credential_apply_policy_fn const policy_credential_apply
+      = credential == NULL ? NULL : credential->_internal.apply_credential_policy;
+
+  if (credential == AZ_CREDENTIAL_ANONYMOUS || policy_credential_apply == NULL)
```

Comment:
Remove the check for Anonymous.  The policy == null check is sufficient b/c anonymous has a null policy.  Should Anoymous ever change to have a poicy function, then this code doesn't need to change.
```suggestion
  if (policy_credential_apply == NULL)
```

Reply:
Anonymous is NULL pointer itself, so we shouldn't read it, and if we try, .policy_credential_apply is most likely going to be initialized with some random data.